lib/rspec-puppet-facts.rb

```diff
@@ -348,4 +343,35 @@ def self.down_facter_version(version, minor_subtractor = 1)
     minor = (minor.to_i - minor_subtractor).to_s
     "#{major}.#{minor}.#{z}"
   end
+
+  def self.facter_version_for_puppet_version(puppet_version)
+    return Facter.version if puppet_version.nil?
+    json_path = File.expand_path(File.join(__dir__, '..', 'ext', 'puppet_agent_components.json'))
+    return Facter.version unless File.file?(json_path) && File.readable?(json_path)
+
+    fd = File.open(json_path, 'rb')
+    data = JSON.parse(fd.read)
+
+    version_map = data.map { |_, versions|
+      if versions['puppet'].nil? || versions['facter'].nil?
+        nil
+      else
+        [Gem::Version.new(versions['puppet']), versions['facter']]
+      end
+    }.compact
+
+    gem_version = Gem::Version.new(puppet_version)
```

Comment:
```suggestion
    puppet_gem_version = Gem::Version.new(puppet_version)
```

Reply:
And next line too but Github doesn't let you do multiline suggestions :/

Reply:
:+1: